src/main/java/bio/terra/service/dataset/DatasetSummary.java

```diff
@@ -91,17 +97,50 @@ public DatasetSummary storage(List<? extends StorageResource<?, ?>> storage) {
         return this;
     }
 
-    public CloudRegion getStorageResourceRegion(CloudResource storageResource) {
-        return storage.stream()
-            .filter(resource -> resource.getCloudResource() == storageResource)
+    public List<BillingProfileModel> getBillingProfiles() {
+        return billingProfiles;
+    }
+
+    public DatasetSummary billingProfiles(List<BillingProfileModel> billingProfiles) {
+        this.billingProfiles = billingProfiles;
+        return this;
+    }
+
+    public BillingProfileModel getDefaultBillingProfile() {
+        return billingProfiles.stream()
+            .filter(b -> b.getId().equals(defaultProfileId))
             .findFirst()
-            .map(StorageResource::getRegion)
-            .orElseThrow(() -> new StorageResourceNotFoundException(
-                String.format("%s could not be found for %s ", storageResource, id)));
+            .orElseThrow();
+    }
+
+    public CloudRegion getStorageResourceRegion(CloudResource storageResource) {
+        return getCloudResourceAttribute(storageResource, StorageResource::getRegion);
     }
 
     public boolean datasetStorageContainsRegion(GoogleRegion region) {
         return storage.stream()
-                .anyMatch(sr -> sr.getRegion().equals(region));
+            .anyMatch(sr -> sr.getRegion().equals(region));
+    }
+
+    public CloudPlatform getStorageCloudPlatform() {
+        // A Dataset should not have both a bucket and a storage account at this point
+        return storage.stream().filter(s -> s.getCloudResource() == GoogleCloudResource.BUCKET ||
+            s.getCloudResource() == AzureCloudResource.STORAGE_ACCOUNT)
+            .findAny()
+            .map(s -> getCloudResourceAttribute(s.getCloudResource(), StorageResource::getCloudPlatform))
+            .orElseThrow();
+    }
+
+    public CloudPlatform getStorageResourceCloudPlatform(CloudResource cloudResource) {
+        return getCloudResourceAttribute(cloudResource, StorageResource::getCloudPlatform);
+    }
+
+    private <T> T getCloudResourceAttribute(CloudResource cloudResource, Function<StorageResource<?, ?>, T> accessor) {
+        return storage.stream()
+            .filter(resource -> resource.getCloudResource() == cloudResource)
```

Comment:
For readability, I think it would help to be consistent when using a name to loop over the objects in `storage.stream()` within this class. In this case, it's called `resource`, above it's written as `s` and `sr`.